api/cpp/docs/mcu/stm32/generic.md

```diff
@@ -12,30 +12,95 @@ as well as the peripherals of our board.
+    
+    # Replace $BSP_NAME with the name of your concrete BSP,
+    # for example stm32h735g_discovery.
+    target_compile_definitions(your-target PRIVATE
+        SLINT_STM32_BSP_NAME=$BSP_NAME
+    )
+    
+    # Link Slint run-time library
+    target_link_libraries(your-target PRIVATE
+        Slint::Slint
+    )
+    ```
+12. Add a C++ source file to your project, for example `appmain.cpp`, with an `appmain` function and call it
     from the generated `main.c`.
-12. Add a `.slint` file to your project and include it in your project by calling `slint_target_sources`.
-13. In your `appmain` function, initialize the screen via `BSP_LCD_InitEx` as well as the touch screen via `BSP_TS_Init`.
-14. Include `#include <slint-stm32.h>` and call `slint_stm32_init(SlintPlatformConfiguration());` to initialize the
-    Slint platform integration for STM32.
-15. Finally, include the header file for your `.slint` file, instantiate the generated class, and invoke the event loop by
-    calling `->run()` on the instance.
+13. Create `app-window.slint` with the following contents:
+    ```slint,no-preview
+    import { VerticalBox, AboutSlint } from "std-widgets.slint";
+    export component AppWindow inherits Window {
+        VerticalBox {
+            AboutSlint {}
+            Text {
+                text: "Hello World";
+                font-size: 18px;
+                horizontal-alignment: center;
+            }
+        }
+    }
+    ```
+14. Add a `.slint` file to your project and include it in your project by calling `slint_target_sources`:
+    ```cmake
+    slint_target_sources(your-target app-window.slint)
+    ```
+15. In your `appmain` function, initialize the screen via `BSP_LCD_InitEx` as well as the touch screen via `BSP_TS_Init`,
+    include `#include <slint-stm32.h>` and call `slint_stm32_init(SlintPlatformConfiguration());` to initialize the
+    Slint platform integration for STM32. Finally, include the header file for your `.slint` file, instantiate the generated class, and invoke the event loop by calling `->run()` on the instance. Use the following example as reference:
+    ```cpp
+    #include <slint-stm32.h>
+    #include <stdio.h>
+    #include <stm32h735g_discovery.h>
+    #include <stm32h735g_discovery_lcd.h>
+    #include <stm32h735g_discovery_ts.h>    
 
+    #include "app-window.h"
+
+    // Called from main()
+    extern "C" void appmain() {
+        if (BSP_LCD_InitEx(0, LCD_ORIENTATION_LANDSCAPE, LCD_PIXEL_FORMAT_RGB565,
+                            LCD_DEFAULT_WIDTH, LCD_DEFAULT_HEIGHT) != 0) {
+            Error_Handler();
+        }
+
+        BSP_LCD_DisplayOn(0);
+        BSP_LCD_SetActiveLayer(0, 0);
+
+        TS_Init_t hTS;
+        hTS.Width = LCD_DEFAULT_WIDTH;
+        hTS.Height = LCD_DEFAULT_HEIGHT;
+        hTS.Orientation = TS_SWAP_XY;
+        hTS.Accuracy = 0;
+        /* Touchscreen initialization */
+        if (BSP_TS_Init(0, &hTS) != 0) {
+            Error_Handler();
+        }
```

Comment:
I wonder how much of this should be done in slint_stm32_init instead.

Reply:
Be my guest :-)